### PR TITLE
PLAT-1161 | Fix coupons syncing

### PR DESCRIPTION
**Pull request by rogatty**
https://bethink.atlassian.net/browse/PLAT-1161

Remember to remove all coupons from LDEK before releasing this and update env on lek production to re-enable syncing.

### CODE REVIEW

File: app/Events/Coupons/CouponEvent.php [hunk 1]
```diff
@@ -2,15 +2,22 @@
...
-abstract class CouponEvent {
+abstract class CouponEvent
+{
 	public $coupon;
 
-	abstract public function shouldSync();
 	abstract public function sync();
 
-	public function issueSyncRequest($method, $coupon) {
+	public function shouldSync(Coupon $coupon)
```
Comment (kvas-damian) [inline app/Events/Coupons/CouponEvent.php]: do we need a separate method for that?

**Comment by kvas-damian** [comment]
Tests are failing

File: app/Events/Coupons/CouponEvent.php (hunk 1 above)
Comment (rogatty) [reply app/Events/Coupons/CouponEvent.php]: We're overriding it in the child classes, so I think it makes sense.

**Comment by bognix** [comment]
@rogatty how about creating another PR with migration that would mark "kind" column as not nullable in database?

**Pull request merged**